pythonscript/_godot_io.pxi

```diff
@@ -0,0 +1,163 @@
+from io import RawIOBase
+from godot._hazmat.conversion cimport (
+    godot_string_to_pyobj,
+    pyobj_to_godot_string,
+    godot_variant_to_pyobj,
+)
+from godot._hazmat.gdnative_api_struct cimport (
+    godot_string,
+    godot_string_name,
+    godot_bool,
+    godot_array,
+    godot_pool_string_array,
+    godot_object,
+    godot_variant,
+    godot_variant_call_error,
+    godot_method_rpc_mode,
+    godot_pluginscript_script_data,
+    godot_pluginscript_instance_data,
+    godot_variant_call_error_error,
+    godot_variant_type
+)
+
+class GodotIOStream(RawIOBase):
+
+    def __init__(self, godot_print_func):
+        self.buffer = ""
+        self.godot_print_func = godot_print_func
+
+    def write(self, b):
+        self.buffer += b
+        if "\n" in self.buffer:
+            *to_print, self.buffer = self.buffer.split("\n")
```

Comment:
`to_print, self.buffer =  self.buffer.rsplit("\n", 1)` ?

Reply:
Done

Reply:
Actually:
`to_print, self.buffer, _ = self.buffer.rsplit("\n", 2)`
(last item is an empty string)

I think split looks cleaner and more intuitive

Reply:
I'm not sure to follow you... if we have `aaa\nbbb\n` we should call `self.godot_print_func("aaa\nbbb")` and if we have `aaaa\nbbbb\ncccc` we should call the same thing but store `cccc` in `self.buffer` ?

But with `aaa\nbbb\n`, ` to_print, self.buffer, _ = self.buffer.rsplit("\n", 2)` gives `aaa` for to_print and `bbb` for buffer (hence `bbb` is not printed yet)
With `aaa\nbbb\nccc` we end up with a similar result and `ccc` is just dropped...

Reply:
You're right, total brainfart!